app/connectors/PayApiConnector.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import com.google.inject.Inject
+import config.ConfigDecorator
+import models.PaymentRequest
+import play.api.http.Status._
+import play.api.libs.json.{Json, Reads}
+import services.http.WsAllMethods
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+final case class CreatePayment(journeyId: String, nextUrl: String)
+
+object CreatePayment {
+  implicit val format = Json.format[CreatePayment]
+}
+
+sealed trait CreatePaymentResponse
+final case class CreatePaymentSuccess(createPaymentResponse: CreatePayment) extends CreatePaymentResponse
+object CreatePaymentFailed extends CreatePaymentResponse
+
+class PayApiConnector @Inject()(http: WsAllMethods, configDecorator: ConfigDecorator) {
```

Comment:
Tried this initially, would need to create a concrete HttpPost for this to work, not sure what more this would gain apart from a lightweight HttpPost. Think most of this goes away with the bootstrap upgrade anyway, as they give us a default http client to work with.

Reply:
:+1: